fix :terminal flickering caused by nonzero scrolloff

LawAbidingCactus opened the pull request:
This change resolves #11072 and #11915.

LawAbidingCactus committed:
terminal: always return zero from get_scrolloff_value() #12230

Fixes flicker

fix #11072
fix #11915

justinmk commented:
@LawAbidingCactus thanks for looking into that. This looks like a good direction, but maybe not the ideal impl. 

Is there a reason this would be preferrable to actually disabling 'scrolloff' during TERM_FOCUS? Then the decision would be local to `terminal.c`, rather than special-cased in `get_scrolloff_value`

justinmk committed:
terminal: disable 'scrolloff' (fixes flicker)

Besides the special-case in get_scrolloff_value(), it makes sense for
'scrolloff' and 'sidescrolloff' to reflect the correct values (for
plugins, scripts, …).

ref 53d607af9c53accfd634435908fb79061f1212b9
ref #11915
ref #12230
